src/java.base/share/classes/java/util/regex/MatchResult.java

```diff
@@ -184,6 +307,49 @@ public interface MatchResult {
      *
      * @return The number of capturing groups in this matcher's pattern
      */
-    public int groupCount();
+    int groupCount();
+
+    /**
+     * Returns an unmodifiable map from capturing group names to group numbers.
+     * If there are no named groups, returns an empty map.
+     *
+     * @return an unmodifiable map from capturing group names to group numbers
+     *
+     * @throws UnsupportedOperationException
+     *          The default implementation of this method always throws
```

Comment:
The `@throws` clause needs to state a general contract over all implementations, so it should say something like, "@throws UOE if the implementation does not support named groups". Then, the specification for the default implementation always throwing UOE should be moved to `@implSpec`.

Reply:
Done